src/snowplow/shareableListItem/shareableListItemEventHandler.integration.ts

```diff
@@ -0,0 +1,174 @@
+import { expect } from 'chai';
+import { ObjectUpdate, EventType, shareableListItemEventSchema } from './types';
+import { ShareableListItemEventHandler } from './shareableListItemEventHandler';
+import {
+  testShareableListItemData,
+  testPartialShareableListItemData,
+} from './testData';
+import {
+  resetSnowplowEvents,
+  getAllSnowplowEvents,
+  getGoodSnowplowEvents,
+  parseSnowplowData,
+} from '../testUtils';
+
+function assertValidSnowplowObjectUpdateEvents(
+  events,
+  triggers: ObjectUpdate['data']['trigger'][]
+) {
+  const parsedEvents = events
+    .map(parseSnowplowData)
+    .map((parsedEvent) => parsedEvent.data);
+
+  expect(parsedEvents).to.include.deep.members(
+    triggers.map((trigger) => ({
+      schema: shareableListItemEventSchema.objectUpdate,
+      data: { trigger: trigger, object: 'shareable_list_item' },
+    }))
+  );
+}
+
+function assertShareableListItemSchema(eventContext) {
+  expect(eventContext.data).to.include.deep.members([
+    {
+      schema: shareableListItemEventSchema.shareable_list_item,
+      data: {
+        shareable_list_item_external_id:
+          testShareableListItemData.shareable_list_item_external_id,
+        shareable_list_external_id:
+          testShareableListItemData.shareable_list_external_id,
+        given_url: testShareableListItemData.given_url,
+        title: testShareableListItemData.title,
+        excerpt: testShareableListItemData.excerpt,
+        image_url: testShareableListItemData.image_url,
+        authors: testShareableListItemData.authors,
+        publisher: testShareableListItemData.publisher,
+        sort_order: testShareableListItemData.sort_order,
+        created_at: testShareableListItemData.created_at,
+        updated_at: testShareableListItemData.updated_at,
+      },
+    },
+  ]);
+}
+
+function assertPartialShareableListItemSchema(eventContext) {
+  expect(eventContext.data).to.include.deep.members([
+    {
+      schema: shareableListItemEventSchema.shareable_list_item,
+      data: {
+        shareable_list_item_external_id:
+          testPartialShareableListItemData.shareable_list_item_external_id,
+        shareable_list_external_id:
+          testPartialShareableListItemData.shareable_list_external_id,
+        given_url: testPartialShareableListItemData.given_url,
+        sort_order: testPartialShareableListItemData.sort_order,
+        created_at: testPartialShareableListItemData.created_at,
+      },
+    },
+  ]);
+}
+
+const testEventData = {
+  shareable_list_item: {
+    ...testShareableListItemData,
+  },
+};
+
+const testPartialEventData = {
+  shareable_list_item: {
+    ...testPartialShareableListItemData,
+  },
```

Comment:
just curious - could this be `shareable_list_item: testPartialShareableListItemData`? is there a reason to spread it into a new object?

Reply:
No reason tbh 😅I think there is a chore ticket to cleanup and refactor some of the code for other event consumers and handlers. I can add this there to take a look at some of the tests and simplify.